mrbelvedereci/build/filters.py

```diff
@@ -0,0 +1,92 @@
+import rest_framework_filters as filters
+from mrbelvedereci.build.models import Build
+from mrbelvedereci.build.models import BuildFlow
+from mrbelvedereci.build.models import Rebuild
+from mrbelvedereci.cumulusci.filters import OrgRelatedFilter
+from mrbelvedereci.cumulusci.filters import ScratchOrgInstanceRelatedFilter
+from mrbelvedereci.cumulusci.models import Org
+from mrbelvedereci.cumulusci.models import ScratchOrgInstance
+from mrbelvedereci.plan.filters import PlanRelatedFilter
+from mrbelvedereci.plan.models import Plan
+from mrbelvedereci.repository.filters import BranchRelatedFilter
+from mrbelvedereci.repository.filters import RepositoryRelatedFilter
+from mrbelvedereci.repository.models import Branch
+from mrbelvedereci.repository.models import Repository
+
+class BuildRelatedFilter(filters.FilterSet):
+    branch = filters.RelatedFilter(
+        BranchRelatedFilter,
+        name='branch',
+        queryset=Branch.objects.all()
+    )
+    org = filters.RelatedFilter(
+        OrgRelatedFilter,
+        name='org',
+        queryset=Org.objects.all()
+    )
+    plan = filters.RelatedFilter(
+        PlanRelatedFilter,
+        name='plan',
+        queryset=Plan.objects.all()
+    )
+    repo = filters.RelatedFilter(
+        RepositoryRelatedFilter,
+        name='repo',
+        queryset=Repository.objects.all()
+    )
+
+    class Meta:
+        model = Build
+        fields = {
+            'commit': ['exact'],
+            'status': ['exact'],
+            'time_queue': ['gt','lt'],
+            'time_start': ['gt','lt'],
+            'time_end': ['gt','lt'],
+        }
+
+class BuildFilter(BuildRelatedFilter):
```

Comment:
Why are these done this way instead of defining directly in BuildFilter?

Reply:
In anticipation that we'll want to have an option for more restricted filters via a relationship than we'd want on the object itself directly.  The main views of each object would use the *Filter while any filter that needs to reference that object would reference its *RelatedFilter.  The wiring is in place for that now so all we need to do to allow more specific filtering on the object itself is override the *Filter class